posthog/hogql_queries/web_analytics/web_overview_pre_aggregated.py

```diff
@@ -0,0 +1,142 @@
+        query = self.runner.query
+
+        # Only works for properties we know are in the pre-aggregated tables
+        for prop in query.properties:
+            if hasattr(prop, "key") and prop.key not in self.SUPPORTED_PROPERTIES:
+                return False
+
+        if query.conversionGoal:
+            return False
+
+        # Only work for fixed-dates that don't include current-date in the filters.
+        today = datetime.now(UTC).date()
+        if self.runner.query_date_range.date_to().date() >= today:
+            return False
+
+        return True
+
+    def get_query(self) -> ast.SelectQuery:
+        current_date_from = self.runner.query_date_range.date_from_str
+        current_date_to = self.runner.query_date_range.date_to_str
+
+        # Handle previous period
+        if self.runner.query_compare_to_date_range:
+            previous_date_from = self.runner.query_compare_to_date_range.date_from_str
+            previous_date_to = self.runner.query_compare_to_date_range.date_to_str
+        else:
+            # If we don't have a previous period, we can just use the same data as the values won't be used
+            # and our query stays simpler.
+            previous_date_from = current_date_from
+            previous_date_to = current_date_from
+
+        # Define date filter conditions first
+        current_period_filter = f"day_bucket >= '{current_date_from}' AND day_bucket <= '{current_date_to}'"
+        previous_period_filter = f"day_bucket >= '{previous_date_from}' AND day_bucket <= '{previous_date_to}'"
+
+        # Define common query patterns for averages over sessions
+        def safe_avg_calc(metric, period_filter):
+            sessions = f"uniqMergeIf(sessions_uniq_state, {period_filter})"
+            return f"""
+            if(
+                {sessions} > 0,
+                sumMergeIf({metric}_state, {period_filter}) / {sessions},
+                0
+            )"""
+
+        query_str = f"""
+        SELECT
+            uniqMergeIf(persons_uniq_state, {current_period_filter}) AS unique_persons,
+            uniqMergeIf(persons_uniq_state, {previous_period_filter}) AS previous_unique_persons,
+
+            sumMergeIf(pageviews_count_state, {current_period_filter}) AS pageviews,
+            sumMergeIf(pageviews_count_state, {previous_period_filter}) AS previous_pageviews,
+
+            uniqMergeIf(sessions_uniq_state, {current_period_filter}) AS unique_sessions,
+            uniqMergeIf(sessions_uniq_state, {previous_period_filter}) AS previous_unique_sessions,
+
+            {safe_avg_calc("total_session_duration", current_period_filter)} AS avg_session_duration,
+            {safe_avg_calc("total_session_duration", previous_period_filter)} AS previous_avg_session_duration,
+
+            {safe_avg_calc("total_bounces", current_period_filter)} AS bounce_rate,
+            {safe_avg_calc("total_bounces", previous_period_filter)} AS previous_bounce_rate,
+
+            NULL AS revenue,
```

Comment:
Huh, that's sad, but that's ok, we'll eventually remove this

Reply:
Oh yeah, don't take it to heart, it is just a POC ❤️